### PR TITLE
Add JWK to PEM & missing trailing slash handler

### PR DESCRIPTION
**Pull request by conor909**
As titled plus tests

### CODE REVIEW

File: test/jwks.spec.js [hunk 1]
```diff
@@ -187,16 +190,34 @@ t.test('calling the clear cache function resets the cache and clears keys', asyn
   t.end()
 })
 
+t.test('If an issuer provides a domain where there is a missing trailing slash, it should be handled', async t => {
```
Comment (simoneb) [inline test/jwks.spec.js:193]: nit: start a test description with a lowercase letter

File: test/jwks.spec.js [hunk 2]
```diff
@@ -187,16 +190,34 @@ t.test('calling the clear cache function resets the cache and clears keys', asyn
...
+  t.end()
+})
+
 const jwks = {
   keys: [
     {
       alg: 'RS512',
       kid: 'KEY',
-      x5c: ['UNUSED']
+      // x5c: ['UNUSED'], // AWS cognito
```
Comment (simoneb) [inline test/jwks.spec.js:208]: remove comment

File: test/jwks.spec.js [hunk 3]
```diff
@@ -187,16 +190,34 @@ t.test('calling the clear cache function resets the cache and clears keys', asyn
...
+  const secret = await getJwks.getSecret({ domain: domainWithMissingTrailingSlash, alg: 'RS512', kid: 'KEY' })
+  t.ok(secret)
+  t.end()
+})
+
 const jwks = {
   keys: [
     {
       alg: 'RS512',
       kid: 'KEY',
```
Comment (simoneb) [inline test/jwks.spec.js:207]: why do we have 2 keys with the same ID? are we sure we're testing both?

**Review by simoneb — COMMENTED** [review]
See comments. Also, can you please review the whole test code to make sure that:

- there are no unused variables: I see a few callback function variables that are clearly unused
- test names make sense: some test names don't make any sense in English

File: test/jwks.spec.js (hunk 3 above)
Comment (conor909) [reply test/jwks.spec.js:207]: Just the `alg` is different, but I can add more variety of tests

Comment (simoneb) [reply test/jwks.spec.js:207]: my point is, if we're trying to reproduce a realistic scenario, we're not doing it. we can't have two keys with the same id

**Review by simoneb — APPROVED** [review]

**Pull request merged**